src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/delay/impl/RetryConsumerDelayStrategy.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.retry.delay.impl;
+
+import org.springframework.beans.factory.annotation.Value;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.retry.delay.ConsumerDelayStrategy;
+
+import java.util.HashMap;
+
+public class RetryConsumerDelayStrategy implements ConsumerDelayStrategy {
+
+    private final ApplicationLogger logger;
+
+    @Value("${RETRY_THROTTLE_RATE_SECONDS}")
+    private long retryThrottleRateSeconds;
+
+    public RetryConsumerDelayStrategy(ApplicationLogger logger) {
+        this.logger = logger;
+    }
+
+    @Override
+    public void throttle() {
+        try {
+            logger.info("Delaying retry consumer for " + retryThrottleRateSeconds + " seconds");
+            long retryThrottleRateMilliseconds = retryThrottleRateSeconds * 1000;
+            Thread.sleep(retryThrottleRateMilliseconds);
+            logger.info("Delay period over");
```

Comment:
To help with logging, you could add a 'consumerId' string in the constructor similar to the LoopingMessageProcessor and pass in the same id value as the consumer it belongs to (see the configuration where we pass an id string into the looping processor). Then we can include this consumerId in teh log messages eg (consumerId + " - Delay period over").  See how it is done in the LoopingMessageProcessor.

Reply:
Added id